userspace/libsinsp/parsers.h

```diff
@@ -193,6 +193,14 @@ class sinsp_parser {
 		return (m_sinsp_mode.is_live() || is_syscall_plugin_enabled()) && m_large_envs_enabled;
 	}
 
+	bool must_notify_thread_user_update() const {
+		return m_sinsp_mode.is_live() || is_syscall_plugin_enabled();
+	}
+
+	bool must_notify_thread_group_update() const {
+		return m_sinsp_mode.is_live() || is_syscall_plugin_enabled();
+	}
+
```

Comment:
I'd expect these two to always be equivalent (we treat users and groups ~the same everywhere). Not sure if we want to keep them separate or merge, your call. Otherwise LGTM, I'll approve once you resolve this comment (one way or another)

Reply:
As the main public APIs `sinsp_threadinfo::set_user()` and `sinsp_threadinfo::set_group()` allow to control the notification independently, I decided to not reduce the granularity level in the other APIs accessing these public helpers (e.g.: `sinsp_threadinfo::init`)